cmd/api/main.go

```diff
@@ -1,433 +1,57 @@
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
-	if itemsTotal == 0 {
-		responseJSON(w, http.StatusOK, []Book{})
-		return
-	}
-
-	pagesTotal, page, pageSize, err := pagination(query, itemsTotal)
-	if err != nil {
-		responseJSON(w, http.StatusBadRequest, err)
-		return
-	}
-
-	//Ask filtered list to db:
-	returnedBooks, err := listBooks(name, minPrice32, maxPrice32, sortBy, sortDirection, archived, page, pageSize)
+func main() {
+	err := run()
 	if err != nil {
 		log.Println(err)
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
-
-	type PagedBooks struct {
-		PageCurrent int    `json:"page_current"`
-		PageTotal   int    `json:"page_total"`
-		PageSize    int    `json:"page_size"`
-		ItemsTotal  int    `json:"items_total"`
-		Results     []Book `json:"results"`
-	}
-
-	pageOfBooksList := PagedBooks{
-		PageCurrent: page,
-		PageTotal:   pagesTotal,
-		PageSize:    pageSize,
-		ItemsTotal:  itemsTotal,
-		Results:     returnedBooks,
-	}
-
-	responseJSON(w, http.StatusOK, pageOfBooksList)
-}
-
-/*Validates and prepares the pagination parameters of the query.*/
-func pagination(query url.Values, itemsTotal int) (pagesTotal, page, pageSize int, err error) {
-
-	pageStr := query.Get("page") //Convert page value to int and set default to 1.
-	if pageStr == "" {
-		page = 1
-	} else {
-		page, err = strconv.Atoi(pageStr)
-		if err != nil {
-			return 0, 0, 0, errResponseQueryPageInvalid
-		}
-		if page <= 0 {
-			return 0, 0, 0, errResponseQueryPageInvalid
-		}
-	}
-
-	pageSizeStr := query.Get("page_size") //Convert page_size value to int and set default to 10.
-	if pageSizeStr == "" {
-		pageSize = 10
-	} else {
-		pageSize, err = strconv.Atoi(pageSizeStr)
-		if err != nil {
-			return 0, 0, 0, errResponseQueryPageInvalid
-		}
-		if !(0 < pageSize && pageSize < 31) {
-			return 0, 0, 0, errResponseQueryPageInvalid
-		}
-	}
-
-	pagesTotal = int(math.Ceil(float64(itemsTotal) / float64(pageSize)))
-	if page > pagesTotal {
-		return 0, 0, 0, errResponseQueryPageOutOfRange
-	}
-
-	return pagesTotal, page, pageSize, nil
-}
-
-/*Validates and prepares the ordering parameters of the query.*/
-func extractOrderParams(query url.Values) (sortBy string, sortDirection string, valid bool) {
-	sortDirection = query.Get("sort_direction")
-	switch sortDirection {
-	case "":
-		sortDirection = "asc"
-	case "asc":
-		break
-	case "desc":
-		break
-	default:
-		return sortBy, sortDirection, false
-	}
-
-	sortBy = query.Get("sort_by")
-	switch sortBy {
-	case "":
-		sortBy = "name"
-	case "name":
-		break
-	case "price":
-		break
-	case "inventory":
-		break
-	case "created_at":
-		break
-	case "updated_at":
-		break
-	default:
-		return sortBy, sortDirection, false
+		os.Exit(1)
 	}
-
-	return sortBy, sortDirection, true
 }
 
-func main() {
+func run() error {
 	//connect to db:
-	dbObject, err := connectDb()
+	connStr := os.Getenv("DATABASE_URL") //I'M STILL CONFUSED ABOUT THAT. IN DATABASE_TEST WE CHANGED THE ENV BY A LITERAL
```

Comment:
We can change the test code to read from the environment too, would be a nice improvement.

Reply:
I took a look at the old commits. now, to remember how we used to deal with the connStr in database_test before.
I've got it:
We used to SET (not GET!!) the connStr as an ENV because ConnectDB() did not receive anything:
`func TestMain(m *testing.M) {
	os.Setenv("DATABASE_URL", "postgres://root:root@localhost:5432/booksdb?sslmode=disable")
	db, err := ConnectDb()`

Now that `database.ConnectDb(connStr)` receive the connStr, we do not need the ENV anymore.

Reply:
Yeah, but its better to control the connection string from the outside.
We can have a quick pairing session for me to show you why is that.